### ISSUE TITLE
Representation of formulas

### ISSUE THREAD

**Issue by gisellemnr** [natvern/CCLITP#7]
How are the formulas represented? If this is implemented in a class, you can change your representation without affecting most of the code. Here's an initial idea for such a class:

```python
from enum import Enum

# name is name, value is arity
class Conn(Enum):
    AND = 2
    OR = 2
    IMPL = 2
    NOT = 1
    NONE = 0 # for atoms

class Formula:
    def __init__(self, connective, args):
        self.conn = connective
        self.sub_forms = args

    def isAnd(self):
        return self.conn == Conn.AND

    def isOr(self):
        return self.conn == Conn.OR

    def isImpl(self):
        return self.conn == Conn.IMPL

    def isNot(self):
        return self.conn == Conn.NOT

    def isAtom(self):
        return self.conn == Conn.NONE

    def getLeftSubformula(self):
        if self.conn.value == 2:
            return self.sub_forms[0]
        else:
            print("Non-binary connective has no left subformula.")

# For testing. Ideally, the list of arguments should be a list of formulas.
f = Formula(Conn.AND, ["A", "B"])
```

**Comment by natvern** [comment]
Started on 3958295d7f2a2988b0615483f868cc310edb7158.
The calculus and engine have been adapted to this new representation (on engine_v2)
Will be completely fixed once I adapt the graphical user interface to it!

**Comment by natvern** [comment]
Fixed on 5514fb43d507ce87366d746d1180694388524340.
